samcli/lib/list/endpoints/endpoints_to_table_mapper.py

```diff
@@ -0,0 +1,65 @@
+"""
+Implementation of the endpoints to table mapper
+"""
+from typing import Dict, Any
+from collections import OrderedDict
+from samcli.lib.list.list_interfaces import Mapper
+
+
+class EndpointsToTableMapper(Mapper):
+    """
+    Mapper class for mapping endpoints data for table output
+    """
+
+    def map(self, data: list) -> Dict[Any, Any]:
+        """
+        Maps data to the format needed for consumption by the table consumer
+
+        Parameters
+        ----------
+        data: list
+            List of dictionaries containing the entries of the endpoints data
+
+        Returns
+        -------
+        table_data: Dict[Any, Any]
+            Dictionary containing the information and data needed for the table consumer
+            to output the data in table format
+        """
+        entry_list = []
+        for endpoint in data:
+            cloud_endpoint_furl_string = endpoint.get("CloudEndpoint", "-")
+            methods_string = "-"
+            cloud_endpoint_furl_multi_list = []
+            if isinstance(endpoint.get("CloudEndpoint", "-"), list) and endpoint.get("CloudEndpoint", []):
+                cloud_endpoint_furl_string = endpoint.get("CloudEndpoint", ["-"])[0]
+                if len(endpoint.get("CloudEndpoint", [])) > 1:
```

Comment:
Could there be an endpoint which is actually in 1 char length (like root path `/` for instance)?
If so, we could change the default value in previous line to be empty string rather than `-`.

Reply:
I don't think there should be a case where the endpoint itself is only a single character (even in the case of customer domains it would still need to be in the format of `a.b`.